### ISSUE TITLE
Review HTTP Abstractions

### ISSUE THREAD

**Issue by glennc** [aspnet/HttpAbstractions#12]
Review how usable the abstractions are. HttpContext, HttpRequest, HttpResponse and family.

**Comment by borgdylan** [comment]
What is the progress of giving access to the remote IP address? As things are one cannot implement anything that needs this information like passing client IP restrictions to payment gateways.

**Comment by Tratcher** [comment]
It's already accessible as shown here:
http://www.tugberkugurlu.com/archive/getting-the-clients-ip-address-in-asp-net-vnext-web-applications
This should unblock any implementation scenarios. This work item is just about making such data more convenient.

**Comment by stephenjwatkins** [comment]
While running in IIS 8, the IP address returned from Tugberk's implementation is incorrect. If we enable IIS 8 to log the requests, we can see the correct client IP address being written to the IIS logs. However, when we log the IP address in our application code (as returned from the implementation above), it is a different IP address.

We also ran an ASP.NET 4.5 application on the same IIS server and noticed that it did return the correct IP address by getting the `Request.UserHostAddress`.

Why is there inconsistency from the IIS logs and the .NET 5 application code retrieving the IIS? How do we get the same IP address that IIS is logging (the correct one)?

**Comment by Tratcher** [comment]
Lets see, on IIS7 UserHostAddress maps to the server variable REMOTE_ADDR. @GrabYourPitchforks any idea why this would be different from the address provided by Http.Sys?

**Comment by Tratcher** [comment]
Ah, @stephenjwatkins are you running behind a reverse proxy? http://stackoverflow.com/questions/19399808/does-iis-with-arr-support-fully-transparent-reverse-proxy

**Comment by stephenjwatkins** [comment]
@Tratcher It's not behind a reverse proxy.

**Comment by Tratcher** [comment]
@stephenjwatkins Please open a new issue for your specific problem. It will get lost if it stays in this one.

**Comment by stephenjwatkins** [comment]
@Tratcher Thanks, opened new issue #181.
